lib/vicadmin/validate.go

```diff
@@ -209,14 +204,33 @@ func NewValidator(ctx context.Context, vch *config.VirtualContainerHostConfigSpe
 		v.DockerPort = fmt.Sprintf("%d", opts.DefaultTLSHTTPPort)
 	}
 
-	err = v.QueryDatastore(ctx, vch, sess)
+	err := v.QueryDatastore(ctx, vch, sess)
 	if err != nil {
 		log.Errorf("Had a problem querying the datastores: %s", err.Error())
 	}
 	v.QueryVCHStatus(vch, sess)
 	return v
 }
 
+func GetVCHName(ctx context.Context, sess *session.Session) string {
+	defer trace.End(trace.Begin(""))
+
+	var vchName = fmt.Sprintf("VCH")
+	self, err := guest.GetSelf(ctx, sess)
+	if err != nil || self == nil {
+		log.Errorf("Unable to get VCH name due to unknown self-reference: %s", err)
+		log.Infof("Setting the VCH name to VCH")
+		return vchName
+	}
+	self2 := vm.NewVirtualMachineFromVM(ctx, sess, self)
+	vchName, err = self2.Name(ctx)
+	if err != nil {
+		log.Infof("Unable to get VCH name: %s", err)
```

Comment:
if err != nil {
    vchName = defaultVchName
...
}

I think it's fine not to return err since the original cod just attempts to get a name and if not present, then just set to "VCH"